File: spring-cloud-gateway-java/src/main/java/org/boki/springcloudgatewayjava/config/GatewayConfig.java
package org.boki.springcloudgatewayjava.config;

import org.springframework.cloud.gateway.route.RouteLocator;
import org.springframework.cloud.gateway.route.builder.RouteLocatorBuilder;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.cors.CorsConfiguration;

@Configuration
public class GatewayConfig {
    @Bean
    public RouteLocator customRouteLocator(RouteLocatorBuilder builder) {
        CorsConfiguration corsConfig = new CorsConfiguration();
        corsConfig.addAllowedMethod("*");
        corsConfig.addAllowedOrigin("*");
        corsConfig.addAllowedHeader("*");

        return builder.routes()
                .route(
                        "patient_route",
                        r -> r.path("/patients/**")
                                .filters(f -> f.rewritePath("/patients/(?<segment>.*)", "/rest/v1/api/patients/${segment}")
                                        .addRequestHeader("apikey", "boki"))
//                                .uri("http://supabase-supabase-kong.supabase.svc.cluster.local:8000/rest/v1"))
                                .uri("http://localhost:9091"))
                .build();
    }
}

File: spring-cloud-gateway/src/main/kotlin/com/boki/springcloudgateway/config/GatewayConfig.kt
package com.boki.springcloudgateway.config

import org.springframework.cloud.gateway.route.RouteLocator
import org.springframework.cloud.gateway.route.builder.RouteLocatorBuilder
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class GatewayConfig {

    @Bean
    fun routes(builder: RouteLocatorBuilder): RouteLocator {
        return builder.routes()
            .route("angular-route") {
                it.header("FE-APP-NAME", "angular")
                    .and()
                    .path("/**")
                    .uri("http://localhost:9091")
            }
            .route("react-route") {
                it.header("FE-APP-NAME", "react")
                    .and()
                    .path("/**")
                    .uri("http://localhost:9092")
            }
            // 여기에 다른 라우트를 추가할 수 있습니다.
            .build()
    }
}
